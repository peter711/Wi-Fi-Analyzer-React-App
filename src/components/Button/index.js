import React from "react";
import styled from "styled-components";

const ButtonWrapper = styled.div`
  button {
    border: none;
    background: none;
    color: #2196f3;
    font-family: 'Lato';
    font-size: 14px;
    font-weight: bold;
    margin: 0 0.5em 0 0.5em;
    padding: 0.4em 1em 0.4em 1em;
    text-transform: uppercase;
  }

  button.primary {
      background-color: #2196f3;
      border-radius: 5px;
      color: white;
  }

  button:hover {
      background-color: grey;
      border-radius: 5px;
      color: white;
      cursor: pointer;
  }
`;

export const Button = ({ text, className, onClick }) => (
  <ButtonWrapper>
    <button className={className} onClick={onClick}>{text}</button>
  </ButtonWrapper>
);

export default Button;
