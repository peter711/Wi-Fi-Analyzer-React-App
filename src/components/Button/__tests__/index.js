import React from "react";
import { shallow } from "enzyme";

import Button from '../';

describe("<Button/>", () => {
    it("should render", () => {
        const component = shallow(<Button/>);
        expect(component).toMatchSnapshot();
    });

    it("should have text", () => {
        const text = "example text";
        const component = shallow(<Button text={text}/>);
        expect(component.find('button').text()).toEqual(text);
        expect(component).toMatchSnapshot();
    });

    it("should call onChange", () => {
        const onClick = jest.fn();
        const component = shallow(<Button onClick={onClick}/>);

        component.find('button').simulate('click');

        expect(onClick).toBeCalled();
    });
});