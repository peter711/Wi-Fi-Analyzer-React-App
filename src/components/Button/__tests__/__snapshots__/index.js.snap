// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should have text 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    text="example text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <button>
        example text
      </button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "example text",
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": "example text",
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <button>
          example text
        </button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "example text",
          "className": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "example text",
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Button/> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <button />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
        "className": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <button />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": undefined,
          "onClick": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
